### DISCUSSION TITLE
Help with PICKCHARS

### DISCUSSION THREAD

**Discussion by TomintheT** [keepassxreboot/keepassxc#7349] [Q&A] [answered]
Hi. New to Keepass/xc. I'm trying to use the auto-type with the PICKCHARS placeholder with the google chrome browser. Just starting with this basic sequence {TAB}{Password}{TAB}{PICKCHARS}. It runs ok but the pickchars dialog box does not pop-up. 

Auto-type is operating ok with some placeholders but not other.

- If I enter this sequence: {CLEARFIELD}{USERNAME}{TAB}{PASSWORD}{ENTER} The username is entered correctly but the password is duplicated in the password field.
- If I enter this sequence: {USERNAME}{TAB}{PASSWORD}{ENTER} The username and password are entered correctly.
- If I enter this sequence: {CLEARFIELD}{USERNAME}{TAB}{PICKCHARDS:Password:c=3}{ENTER} I get a pop-up that the syntax is incorrect.

Any help appreciated.

**Comment by droidmonkey** [comment, marked as answer]
You need to use snapshot build https://snapshot.keepassxc.org until 2.7.0 is released

**Reply by TomintheT** [reply]
Thanks, have now installed that version but get the following message on attempting to save the sequence:

'An error occurred while validating the custom Auto-Type sequence:
Invalid placeholder: {PICKCHARS:Password:C=5}
Would you like to correct it?'

**Reply by droidmonkey** [reply]
That is not a valid sequence. You can only put {pickchars}

**Reply by TomintheT** [reply]
OK, so can I pick characters from an attribute rather than the password? Like {PICKCHARS:memeroable}

Is there an overview of what placeholders can be used? I was going from the Keepass guidance?

**Reply by TomintheT** [reply]
This is what I'm trying to auto-type to. So picking characters from a string but one character at a time. 

Keypass has something like this:

{PICKCHARS:Password:ID=1, C=1, Conv=D, Conv-Offset=1}{TAB}{PICKCHARS:Password:ID=2, C=1, Conv=D, Conv-Offset=1}{TAB}{PICKCHARS:Password:ID=3, C=1, Conv=D, Conv-Offset=1}{ENTER}

![image](https://user-images.githubusercontent.com/98559540/151575959-990c9e58-266e-49f4-bcb1-4d2aa37896b4.png)

**Reply by droidmonkey** [reply]
You can only pickchars from your password. We don't support any other option at this time.

**Reply by TomintheT** [reply]
OK, thanks very much for your help.

**Reply by r3times** [reply]
> You can only pickchars from your password. We don't support any other option at this time.

Are there any plans to extend use of pickchars to fields other than the password?

**Reply by droidmonkey** [reply]
No
